### PR TITLE
Create new reminders

### LINKED ISSUE #5: Create A New Reminder
At the bottom of the `reminders.hbs` template, there should be a button that routes the user to `'/new'` where users can create a new reminder.

* The new reminder should have inputs for the title, date, and notes.
* The new reminder should have a button for submitting the new note.
* Pressing the button, should submit reminder to the database and have it shown in the Reminders list.

**Hot Tip**: TDD is really helpful.

### PR DESCRIPTION
**Pull request by gprocell927**
🍨 🍫🍨 🍫  @Tman22 , @martensonbj , @nfosterky  🍨 🍫🍨 🍫
## Purpose

Adds a button to the bottom of the `reminders.hbs` template that routes the user to `'/new'` where the user can create a new reminder. Closes #5 

## Approach
This change adds a `new-reminder` component, a `new` route (to allow the user to make a new reminder,  and a `new-reminder` template. 

### Learning

Researched heavily with Ember documentation in regards to DS Store, setting properties and creating records.  [Ember docs](http://emberjs.com/api/data/classes/DS.Store.html#method_createRecord)
Also referenced the Ember groceries tutorial for learning set up for when a user would like to add input [Ember Groceries Repo](https://github.com/turingschool-examples/ember-groceries/tree/master)

### Test coverage 
Acceptance tests were written to test that: </br>
* Clicking the new reminder button would redirect the user to /new
* The /new URL should show a form to submit a new reminder
* When the user inputs a new reminder and clicks the 'submit' button, the new reminder is shown in the reminders area.

### Follow-up tasks

- [Issue #7 - Display Default Page](https://github.com/turingschool-projects/1608-remember-5/issues/7)

### Screenshots
THE ACTION!

### CODE REVIEW

File: app/router.js [hunk 1]
```diff
@@ -8,8 +8,11 @@ const Router = Ember.Router.extend({
 
 Router.map(function() {
   this.route('reminders', function() {
-    this.route('reminder', { path: ':reminder_id' });
+    this.route('reminder', { path: ':reminder_id'
+    });
```
Comment (nfosterky) [inline app/router.js]: These closing brackets should be on the previous line

File: app/router.js [hunk 2]
```diff
@@ -8,8 +8,11 @@ const Router = Ember.Router.extend({
...
+    this.route('new');
   });
+
```
Comment (nfosterky) [inline app/router.js]: Remove this extra line

File: app/templates/reminders.hbs [hunk 3]
```diff
@@ -10,4 +10,7 @@
   </ul>
 </section>
 
+{{#link-to 'reminders.new'}}<button
+  class='new-reminder-button'>New Reminder</button>{{/link-to}}
```
Comment (nfosterky) [inline app/templates/reminders.hbs]: The formatting for this link/button could be better. 

If there is not enough room to fit all of an element on one line you should nest inner elements and indent them.
e.g. 
`<div>`
--`<button class='some-class'>`
----`My Button Text`
--`</button>`
`</div>`

This separates the different elements and makes the code easier to read

File: app/templates/components/new-reminder.hbs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<h2>New Reminder</h2>
+
+<form class="spec-new-reminder--form" {{action 'createReminder' on="submit"}}>
+  <label>
+    Title:
+    {{input value=title class="spec-input-title"}}
+  </label>
+  <label>
+    Date:
+    {{input value=date class="spec-input-date"}}
```
Comment (nfosterky) [inline app/templates/components/new-reminder.hbs]: Use the type attribute for input html elements. 

"How an `<input>` works varies considerably depending on the value of its type attribute."
[HTML Input Element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input)

File: tests/acceptance/reminder-list-test.js [hunk 5]
```diff
@@ -34,4 +34,36 @@ test('clicking on an individual item', function(assert) {
...
+    andThen(function(){
+      assert.equal(Ember.$('spec-new-reminder--form').length, 1)
+    });
+  });
+
+  test('should show new reminder on submit', function (assert) {
+    visit('/reminders/new');
+
+    fillIn('.spec-input-title', 'Bombs')
+    fillIn('.spec-input-date', 'Over')
```
Comment (nfosterky) [inline tests/acceptance/reminder-list-test.js]: Does this work? Doesn't seem like a valid date...

File: tests/acceptance/reminder-list-test.js [hunk 6]
```diff
@@ -34,4 +34,36 @@ test('clicking on an individual item', function(assert) {
...
+    fillIn('.spec-input-notes', 'Baghdad')
+
+    click('.new-reminder--submit')
+
+    andThen(function() {
+      assert.equal(find('h1').text(), 'Bombs', 'should list new reminder name')
```
Comment (nfosterky) [inline tests/acceptance/reminder-list-test.js]: Would love to see tests for the date and notes field

**Comment by gprocell927** [comment]
Previously requested changes have now been made to this branch to close #5.
🍣 🍙 🍣 🍙 🍣 🍙

File: app/templates/reminders.hbs [hunk 7]
```diff
@@ -10,4 +10,10 @@
   </ul>
 </section>
 
+{{#link-to 'reminders.new'}}
+  <button class='new-reminder-button'>
+    New Reminder
+  </button>
+{{/link-to}}
```
Comment (nfosterky) [inline app/templates/reminders.hbs]: 👍

File: tests/acceptance/reminder-list-test.js (hunk 6 above)
Comment (maiastone) [inline tests/acceptance/reminder-list-test.js]: Please review changes and additional tests.  Closes #5

**Comment by maiastone** [comment]
Check out changes and additional tests are passing.  Closes #5

File: app/routes/new.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+  model() {
+    return this.get('store').findAll('new-reminder--form');
```
Comment (nfosterky) [inline app/routes/new.js]: This is unnecessary. findAll looks in the data store for all records for the given type.

http://emberjs.com/api/data/classes/DS.Store.html#method_findAll

**Comment by gprocell927** [comment]
Made and pushed change from `findAll` to `find` in 'new' route. 👺 👹 👺 👹 
Closes #5

**Comment by gprocell927** [comment]
Ok, after some clarification, we have removed the `new` route! 🐇

**Pull request merged**